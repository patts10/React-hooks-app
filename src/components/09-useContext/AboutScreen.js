import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

  const { user, setUser } = useContext(UserContext);

  const handleClick = () => {
    setUser({});
  }

  return (
    <>
     <h1>AboutScreen</h1> 
     <hr />

     <pre>
       {JSON.stringify(user, null, 3)}
     </pre>
     
     <button
      className='bg-yellow-500 shadow-lg shadow-blue-500/50 rounded py-1 px-2 mt-3 text-white'
      onClick={ handleClick }
     >
       Logout
     </button>
    </>
  )
}
