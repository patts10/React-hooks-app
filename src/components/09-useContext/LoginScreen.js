import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

  const { setUser } = useContext(UserContext);

  const user = {
    id: 123,
    name: "Patts"
  }

  return (
    <>
     <h1>LoginScreen</h1> 
     <hr />

     <button
      className='bg-blue-500 shadow-lg shadow-blue-500/50 rounded py-1 px-2 mt-3 text-white'
      onClick={ () => setUser(user) }
     >
       Login
     </button>
     
    </>
  )
}
