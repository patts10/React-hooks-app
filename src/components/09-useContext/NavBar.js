import React from "react";
import { Link, NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav className="relative w-full flex flex-wrap items-center justify-between py-3 bg-gray-900 text-gray-200 shadow-lg navbar navbar-expand-lg navbar-light">
      <div className="container-fluid w-full flex flex-wrap items-center justify-between px-6">
        <div
          className="collapse navbar-collapse flex flex-row items-center"
          id="navbarSupportedContent1"
        >
          <Link className="text-xl text-white pr-2 font-semibold" to="/">
            useContext
          </Link>
          <ul className="navbar-nav flex flex-row pl-0 list-style-none mr-auto">
            <li className="nav-item p-2">
              <NavLink
                className={({ isActive }) => 'nav-item ' + (isActive ? 'active' : 'opacity-60 hover:opacity-90 focus:opacity-80 p-0')}
                to="/"
              >
                Home
              </NavLink>
            </li>
            <li className="nav-item p-2">
              <NavLink 
                className={({ isActive }) => 'nav-item ' + (isActive ? 'active' : 'opacity-60 hover:opacity-90 focus:opacity-80 p-0')}
                to='/login'
              >
                  Login
              </NavLink> 
            </li>
            <li className="nav-item p-2">
              <NavLink
                className={({ isActive }) => 'nav-item ' + (isActive ? 'active' : 'opacity-60 hover:opacity-90 focus:opacity-80 p-0')}
                to="/about"
              >
                About
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};
