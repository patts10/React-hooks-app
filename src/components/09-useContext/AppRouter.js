import React from "react";
import {
  BrowserRouter as Router,
  Routes,
  Route,
  Navigate,
} from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />

        <div className="ml-5">
          <Routes>
            <Route exact path="/" element={<HomeScreen />}></Route>
            <Route exact path="/about" element={<AboutScreen />}></Route>
            <Route exact path="/login" element={<LoginScreen />}></Route>

            <Route path="*" element={<Navigate to={"/"} />}></Route>
          </Routes>
        </div>
      </div>
    </Router>
  );
};
