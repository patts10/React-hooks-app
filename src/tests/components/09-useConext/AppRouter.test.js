import { mount } from 'enzyme';
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import { UserContext } from "../../../components/09-useContext/UserContext";


describe('Tests on <AppRouter />', () => { 

  const user = {
    id: 123,
    name: 'Patts'
  }
  
  const wrapper = mount(
    <UserContext.Provider value={ {user} } >
      <AppRouter />
    </UserContext.Provider>
  );

  test('should show correctly', () => { 
    
    expect(wrapper).toMatchSnapshot();
   })
  
 });