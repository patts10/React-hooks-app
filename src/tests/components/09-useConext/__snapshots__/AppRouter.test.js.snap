// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AppRouter /> should show correctly 1`] = `
<div>
  <nav
    className="relative w-full flex flex-wrap items-center justify-between py-3 bg-gray-900 text-gray-200 shadow-lg navbar navbar-expand-lg navbar-light"
  >
    <div
      className="container-fluid w-full flex flex-wrap items-center justify-between px-6"
    >
      <div
        className="collapse navbar-collapse flex flex-row items-center"
        id="navbarSupportedContent1"
      >
        <a
          className="text-xl text-white pr-2 font-semibold"
          href="/"
          onClick={[Function]}
        >
          useContext
        </a>
        <ul
          className="navbar-nav flex flex-row pl-0 list-style-none mr-auto"
        >
          <li
            className="nav-item p-2"
          >
            <a
              aria-current="page"
              className="nav-item active"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item p-2"
          >
            <a
              className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li
            className="nav-item p-2"
          >
            <a
              className="nav-item opacity-60 hover:opacity-90 focus:opacity-80 p-0"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <div
    className="ml-5"
  >
    Array [
      <h1>
        HomeScreen
      </h1>,
      <hr />,
      <pre>
        {
   "id": 123,
   "name": "Patts"
}
      </pre>,
    ]
  </div>
</div>
`;
